master/internal/db/postgres_rp_workspace_bindings_intg_test.go

```diff
@@ -96,10 +97,136 @@ func TestListWorkspacesBindingRP(t *testing.T) {
 	return
 }
 
-func TestListRPsBoundToWorkspace(t *testing.T) {
-	// pretty straightforward
-	// don't list binding that are invalid
-	// return unbound pools too (we pull from config)
+func TestListRPsAvailableToWorkspace(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(RootFromDB))
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, MigrationsFromDB)
+	ctx := context.Background()
+	user := RequireMockUser(t, db)
+
+	workspaceNames := []string{"workspace1", "workspace2"}
+	workspaceIDs, err := MockWorkspaces(workspaceNames, user.ID)
+	require.NoError(t, err)
+	existingPools := []config.ResourcePoolConfig{
+		{PoolName: "poolName1"},
+		{PoolName: "poolName2"},
+		{PoolName: "poolName3"},
+	}
+
+	// A workspace has no bounded RP
+	rpNames, _, err := ReadRPsAvailableToWorkspace(ctx, workspaceIDs[0], 0, 500, existingPools)
+	require.NoError(t, err)
+
+	expectedRPNames := []string{}
+	for _, rpName := range existingPools {
+		expectedRPNames = append(expectedRPNames, rpName.PoolName)
+	}
+	sort.Strings(rpNames)
+	require.Equal(t, expectedRPNames, rpNames)
+
+	// Workspace 1 has no bounded RP and a RP is bound to workspace 2
+	err = AddRPWorkspaceBindings(ctx, workspaceIDs[1:2], "poolName2", existingPools)
+	require.NoError(t, err)
+
+	rpNames, _, err = ReadRPsAvailableToWorkspace(ctx, workspaceIDs[0], 0, 500, existingPools)
+	require.NoError(t, err)
+	expectedRPNames = []string{"poolName1", "poolName3"}
+	sort.Strings(rpNames)
+	require.Equal(t, expectedRPNames, rpNames)
+
+	err = RemoveRPWorkspaceBindings(ctx, workspaceIDs[1:2], "poolName2")
+	require.NoError(t, err)
+
+	// A workspace has some bounded RP and some unbound RP
+	err = AddRPWorkspaceBindings(ctx, workspaceIDs[0:1], "poolName1", existingPools)
+	require.NoError(t, err)
+	err = AddRPWorkspaceBindings(ctx, workspaceIDs[1:2], "poolName2", existingPools)
+	require.NoError(t, err)
+
+	rpNames, _, err = ReadRPsAvailableToWorkspace(ctx, workspaceIDs[0], 0, 500, existingPools)
+	require.NoError(t, err)
+	expectedRPNames = []string{"poolName1", "poolName3"}
```

Comment:
I [was able to loop over](https://github.com/determined-ai/determined/pull/7176/files#diff-064a2ebca5da57251013259b707bb39017043d342046879c003e9327f88583b5R131) the `existingPools` to get `expectedRPNames`. Not sure how to do the same here and in the other tests.

Reply:
that should be fine, this is only a test anyways so don't bother if its too inconvenient